Framework/API/src/WorkspaceHistory.cpp

```diff
@@ -362,16 +401,24 @@ WorkspaceHistory::parseAlgorithmHistory(const std::string &rawData) {
   // Get the execution date/time
   std::string date, time;
   getWordsInString(info[EXEC_TIME], dummy, dummy, date, time);
-  Poco::DateTime start_timedate;
-  // This is needed by the Poco parsing function
-  int tzdiff(-1);
   Mantid::Types::Core::DateAndTime utc_start;
-  if (!Poco::DateTimeParser::tryParse("%Y-%b-%d %H:%M:%S", date + " " + time,
-                                      start_timedate, tzdiff)) {
-    g_log.warning() << "Error parsing start time in algorithm history entry."
-                    << "\n";
-    utc_start = Types::Core::DateAndTime::defaultTime();
+  // If not legacy version construct normally else Parse in the legacy data
+  if (std::isdigit(date[6])) {
+    Mantid::Types::Core::DateAndTime timeConstruction(date + "T" + time);
+    utc_start = timeConstruction;
+  } else {
+    Poco::DateTime start_timedate;
+    // This is needed by the Poco parsing function
+    int tzdiff(-1);
+    if (!Poco::DateTimeParser::tryParse("%Y-%b-%d %H:%M:%S", date + " " + time,
+                                        start_timedate, tzdiff)) {
```

Comment:
Is the time string supposed to be ISO8601? Would `Mantid::Types::Core::DateTimeHelpers::stringIs*` functions be of any help here?

Reply:
For compatibility reasons, then.